packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -68,6 +68,12 @@ enum _ScaffoldSlot {
 /// [BuildContext] via [ScaffoldMessenger.of] and use the
 /// [ScaffoldMessengerState.showSnackBar] function.
 ///
+/// When the [ScaffoldMessenger] has nested [Scaffold] descendants, the
+/// ScaffoldMessenger will only present a [SnackBar] to the root Scaffold of
+/// the subtree of Scaffolds. In order to show SnackBars in the inner, nested
+/// Scaffolds, set a new scope for your SnackBars by instantiating a new
+/// ScaffoldMessenger in between the levels of nesting.
```

Comment:
optional: may be nice to show a code sample of that here.

Reply:
What do you think of linking to an example in the gallery? I think this sample would be very large, and I don't want to encourage nesting scaffolds... there is a section in the Scaffold docs that says nested Scaffolds is not typically necessary. There are a few cases where it is relevant - like in the gallery where there is embedded Flutter content, but the Scaffold docs recommend avoiding nesting.